### PR TITLE
Support for GCP KMS configuration

### PR DESCRIPTION
**Pull request by Alevsk**
Added support for configuring Tenant encryption via Google Cloud
Platform KMS

### CODE REVIEW

**Review by dvaldivia — DISMISSED** [review]

**Comment by dvaldivia** [comment]
make assets kudasai

**Review by dvaldivia — DISMISSED** [review]

**Comment by harshavardhana** [comment]
> make assets kudasai

For me -- onegaishimasu tamales

File: restapi/admin_tenants_helper.go [hunk 1]
```diff
@@ -489,6 +489,25 @@ func createOrReplaceKesConfigurationSecrets(ctx context.Context, clientSet K8sCl
...
 			}
 		}
+	} else if encryptionCfg.Gcp != nil {
+		// Initialize GCP
+		kesConfig.Keys.Gcp = &kes.Gcp{
+			SecretManager: &kes.GcpSecretManager{},
+		}
+		// GCP basic kesConfiguration
+		if encryptionCfg.Gcp.Secretmanager != nil {
+			kesConfig.Keys.Gcp.SecretManager.ProjectID = *encryptionCfg.Gcp.Secretmanager.ProjectID
```
Comment (cesnietor) [inline restapi/admin_tenants_helper.go:499]: can encryptionCfg.Gcp.Secretmanager.ProjectID be nil? might cause issues when dereferencing.

Comment (Alevsk) [reply restapi/admin_tenants_helper.go:499]: `encryptionCfg.Gcp.Secretmanager.ProjectID` is always mandatory @cesnietor

**Review by dvaldivia — APPROVED** [review]

**Comment by Alevsk** [comment]
> make assets kudasai

done

**Review by dvaldivia — APPROVED** [review]

**Review by bexsoft — APPROVED** [review]

**Pull request merged**